testing/e2e/pkg/testnet.go

```diff
@@ -0,0 +1,315 @@
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"net"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	"github.com/tendermint/tendermint/rpc/client/http"
+)
+
+const (
+	secp256k1Type               = "secp256k1"
+	ed25519Type                 = "ed25519"
+	networkIPv4                 = "10.186.73.0/24"
+	firstProxyPort       uint32 = 4201
+	dockerSrcURL                = "ghcr.io/celestiaorg/celestia-app"
+	randomSeed           int64  = 589308084734268
+	defaultAccountTokens        = 1e6
+)
+
+type Testnet struct {
+	Name     string // also used as the chain-id
+	Dir      string
+	IP       *net.IPNet
+	Nodes    map[string]*Node
+	Accounts map[string]*Account
+}
+
+type Node struct {
+	Name           string
+	Versions       []string
+	StartHeight    int64
+	Peers          []string
+	SignerKey      crypto.PrivKey
+	NetworkKey     crypto.PrivKey
+	AccountKey     crypto.PrivKey
+	IP             net.IP
+	ProxyPort      uint32
+	SelfDelegation int64
+}
+
+type Account struct {
+	Name   string
+	Tokens int64
+	Key    crypto.PrivKey
+}
+
+func LoadTestnet(manifest Manifest, file string) (*Testnet, error) {
```

Comment:
[nit-optional] I think `filePath` is closer to what this parameter means.
```suggestion
func LoadTestnet(manifest Manifest, filePath string) (*Testnet, error) {
```